src/SIL.Harmony/Changes/ChangeContext.cs

```diff
@@ -1,18 +1,25 @@
+using SIL.Harmony.Db;
+
 namespace SIL.Harmony.Changes;
 
 public class ChangeContext : IChangeContext
 {
     private readonly SnapshotWorker _worker;
     private readonly CrdtConfig _crdtConfig;
 
-    internal ChangeContext(Commit commit, SnapshotWorker worker, CrdtConfig crdtConfig)
+    internal ChangeContext(Commit commit, int commitIndex, IDictionary<Guid, ObjectSnapshot> intermediateSnapshots, SnapshotWorker worker, CrdtConfig crdtConfig)
     {
         _worker = worker;
         _crdtConfig = crdtConfig;
         Commit = commit;
+        CommitIndex = commitIndex;
+        IntermediateSnapshots = intermediateSnapshots;
     }
 
-    public CommitBase Commit { get; }
+    CommitBase IChangeContext.Commit => Commit;
+    public Commit Commit { get; }
+    public int CommitIndex { get; }
+    public IDictionary<Guid, ObjectSnapshot> IntermediateSnapshots { get; }
```

Comment:
these new properties should all be internal so Change code doesn't depend on them, especially the snapshots property. Maybe we can make ChangeContext internal? Probably not though.

Reply:
It seems to me that ChangeContext can be internal 🤔